src/Tooltip/Tooltip.tsx

```diff
@@ -53,68 +83,125 @@ export class Tooltip extends React.Component<TooltipProps, TooltipState> {
         }
     }
 
-    /** Forces the tooltip to show */
-    forceShow() {
-        !this.state.toggle && this.setState({ toggle: true });
-    }
-
-    isPositioned(search: string): boolean {
-        const position: string = this.props.position ? this.props.position : "bottom";
-        return position.search(search) === 0;
+    /**
+     * Forces the tooltip to show
+     */
+    forceShow = (): void => {
+        !this.state.visible && this.onTooltipToggle(null, true);
     }
 
-    toggleTooltip(state?: boolean, e?: React.MouseEvent<HTMLDivElement>) {
-        if (state !== undefined) {
-            this.setState({ toggle: state });
-        } else {
-            this.setState({ toggle: !this.state.toggle });
+    onTooltipContentBlur = (e: React.FocusEvent<HTMLDivElement>): void => {
+        const triggeredNode: Node = e.relatedTarget as Node || document.activeElement;
+        if (this.props.trigger !== "click" || !document.getElementById(this.state.referenceId).contains(triggeredNode)) {
+            this.onTooltipToggle(e, false);
         }
-
-        this.props.onClick && this.props.onClick(e);
     }
-
+    // TODO: remove customSvg when attribute is removed
     render() {
         return (
-            <div className={"tooltip-container" + (this.props.className ? ` ${this.props.className}` : "")} id={this.props.id}>
+            <div
+                className={"tooltip-container" + (this.props.className ? ` ${this.props.className}` : "")}
+                id={this.props.id}
+            >
                 <div
-                    className="icon tooltip-icon"
-                    onClick={(e: React.MouseEvent<HTMLDivElement>) => !this.props.triggerOnHover && this.toggleTooltip(undefined, e)}
-                    onMouseEnter={() => this.props.triggerOnHover && this.toggleTooltip(true)}
-                    onMouseLeave={() => this.props.triggerOnHover && this.toggleTooltip(false)}
+                    id={this.state.referenceId}
+                    ref={this.containerRef}
+                    className="tooltip-reference"
+                    tabIndex={-1}
+                    onClick={(e: React.MouseEvent<HTMLDivElement>) => ((!this.props.trigger && !this.props.triggerOnHover) || this.props.trigger === "click") && this.onTooltipToggle(e)}
+                    onMouseEnter={(e: React.MouseEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, true)}
+                    onMouseLeave={(e: React.MouseEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, false)}
+                    onTouchStart={(e: React.TouchEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, true)}
+                    onTouchEnd={(e: React.TouchEvent<HTMLDivElement>) => (this.props.triggerOnHover || this.props.trigger === "hover") && this.onTooltipToggle(e, false)}
```

Comment:
Avoid arrow functions inside the render cycle of class component , especially when they are many. read [here](https://www.freecodecamp.org/news/why-arrow-functions-and-bind-in-reacts-render-are-problematic-f1c08b060e36/)

So instead, try

`onClick={this.handleClick}` 

```
handleClick(e: React.MouseEvent<HTMLDivElement>) {
((!this.props.trigger && !this.props.triggerOnHover) || this.props.trigger === "click") && this.onTooltipToggle(e);
}
```

Reply:
updated